types.go

```diff
@@ -1242,3 +1242,29 @@ type FirewallIPSet struct {
 	Digest  string `json:"digest,omitempty"`
 	Comment string `json:"comment,omitempty"`
 }
+
+type (
+	ModeType     = string
+	CompressType = string
+)
+
+const (
+	ModeSnapshot = ModeType("snapshot")
+	ModeSuspend  = ModeType("suspend")
+	ModeStop     = ModeType("stop")
+
+	CompressZero = CompressType("0")
+	CompressOne  = CompressType("1")
+	CompressGzip = CompressType("gzip")
+	CompressLzo  = CompressType("lzo")
+	CompressZstd = CompressType("zstd")
+)
+
+type VirtualMachineBackupOptions struct {
+	VMID     uint64       `json:"vmid"`
+	Storage  string       `json:"storage"`
+	Remove   int          `json:"remove,omitempty"`
+	Mode     ModeType     `json:"mode,omitempty"`
+	Compress CompressType `json:"compress,omitempty"`
+	Notes    string       `json:"notes,omitempty"`
+}
```

Comment:
I'm not really one to talk here as I've definitely not finished param types before but if you could finish the params for this one I would appreciate it as it's an important feature and we should probably do it the first time. https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/vzdump note

Reply:
Yes, I will finish it